python/paddle/v2/fluid/transpilers/multi_devices.py

```diff
@@ -0,0 +1,148 @@
+from ..framework import Variable, Program, Block, Operator, unique_name
+from .. import core
+
+__all__ = ['transpile_to_multi_devices']
+
+
+def _vars_to_names_(vars):
+    if vars is None:
+        vars = []
+    for each_var in vars:
+        if isinstance(each_var, Variable):
+            yield str(each_var.name)
+        elif isinstance(each_var, basestring):
+            yield str(each_var)
+        else:
+            raise TypeError("Vars should be Variable or basestring")
+
+
+def _get_inputs_and_params_names_(block):
+    outputs = set()
+    inputs_and_params = set()
+    for op in block.ops:
+        assert isinstance(op, Operator)
+        for i_name in op.input_names:
+            for i_var_name in op.input(i_name):
+                if i_var_name not in outputs:
+                    inputs_and_params.add(str(i_var_name))
+
+        for o_name in op.output_names:
+            for o_var_name in op.output(o_name):
+                outputs.add(str(o_var_name))
+    return inputs_and_params
+
+
+def _move_copy_var_to_block_(var_names, src_block, dst_block, remove=True):
+    for p_name in var_names:
+        p_var = src_block.vars.get(p_name, None)
+        if p_var is None:
+            raise ValueError(
+                "cannot move parameter up to dst_block. Variable %s not found",
+                p_name)
+        assert p_name not in dst_block.vars
+        dst_block.clone_var(p_var)
+        if remove:
+            src_block.remove_var(p_var)
+
+
+def _move_var_to_block_(var_names, src_block, dst_block):
+    _move_copy_var_to_block_(var_names, src_block, dst_block, remove=True)
+
+
+def _copy_var_to_block_(var_names, src_block, dst_block):
+    _move_copy_var_to_block_(var_names, src_block, dst_block, remove=False)
+
+
+def _var_lists_(var_names, block):
+    return [block.var(var_name) for var_name in var_names]
+
+
+def transpile_to_multi_devices(program,
```

Comment:
We have several transpilers, are we going to all implement as a function or add some base class?